src/pages/HomePage/HomePage.tsx

```diff
@@ -216,7 +222,7 @@ const HomePage = () => {
               </TableBody>
             </BlockListTable>
           </Content>
-          <PageSwitch totalNumberOfElements={blockList?.total} elementsPerPage={blockList?.blocks?.length} />
+          {blockList?.total && <PageSwitch totalNumberOfElements={blockList.total} elementsPerPage={8} />}
```

Comment:
@mvaivre I purposfully set this to `8`. Without doing so, when we navigate to the last page there is the case that the length of the `blockList?.blocks` will be less than 8. Since the calculation of the `totalNumberOfPages` depends upon this number, I think it makes sense to always have it fixed.

Reply:
Also, there is no reason to render the pagination switch unless we have `blockList?.total`